backend/de.metas.business/src/main/java/de/metas/shipping/PurchaseOrderToShipperTransportationService.java

```diff
@@ -53,14 +73,81 @@ public void addPurchaseOrdersToShipperTransportation(@NonNull final ShipperTrans
 				.create()
 				.idsAsSet(OrderId::ofRepoId)
 				.stream()
-				.filter(orderId -> repo.purchaseOrderNotInShipperTransportation(orderId))
+				.filter(repo::purchaseOrderNotInShipperTransportation)
 				.collect(ImmutableList.toImmutableList());
 
 		for (final OrderId purchaseOrderId : validPurchaseOrdersIds)
 		{
-			repo.addPurchaseOrderToShipperTransportation(purchaseOrderId, shipperTransportationId);
+			addPurchaseOrderToShipperTransportation(purchaseOrderId, shipperTransportationId);
 		}
+	}
 
+	public void addPurchaseOrderToCurrentShipperTransportation(final @NonNull I_C_Order purchaseOrder)
+	{
+		shipperTransportationDAO.getOrCreate(CreateShipperTransportationRequest.builder()
+				.shipperId(ShipperId.ofRepoId(purchaseOrder.getM_Shipper_ID()))
+				.orgId(OrgId.ofRepoId(purchaseOrder.getAD_Org_ID()))
+				.assignAnonymouslyPickedHUs(true)
+				.shipDate(TimeUtil.asLocalDate(purchaseOrder.getDatePromised()))
+				.shipperBPartnerAndLocationId(BPartnerLocationId.ofRepoId(BPartnerId.ofRepoId(purchaseOrder.getC_BPartner_ID()), purchaseOrder.getC_BPartner_Location_ID()))
+				.build());
+	}
+
+	public void addPurchaseOrderToShipperTransportation(final @NonNull OrderId purchaseOrderId, final @Nullable ShipperTransportationId shipperTransportationId)
+	{
+		final org.compiere.model.I_C_Order order = orderDAO.getById(purchaseOrderId);
+
+		final ShipperTransportationId shipperTransportationIdToUse = shipperTransportationId != null ? shipperTransportationId :
+				shipperTransportationDAO.retrieveNextOpenShipperTransportationIdOrNull(ShipperId.ofRepoId(order.getM_Shipper_ID()), TimeUtil.asLocalDate(order.getDatePromised()));
+
+		final List<I_C_OrderLine> orderLines = orderDAO.retrieveOrderLines(purchaseOrderId);
+		final List<I_C_OrderLine> orderLinesWithLUQty = orderLines.stream()
+				.filter(this::isLUQtySet)
+				.collect(Collectors.toList());
+		final boolean isOrderLinesWithoutLUQtyExist = orderLines.stream().anyMatch(ol -> !orderLinesWithLUQty.contains(ol));
```

Comment:
i think u can simply compare the sizes of orderLines and orderLinesWithLUQty.... isn't it?

Reply:
yup